### PR TITLE
Added test for dsolve TypeError: doit()

### LINKED ISSUE #9446: dsolve TypeError: doit() got an unexpected keyword argument 'manual'
It appears there is a regression in `dsolve`. I see the following under `0.7.6` and the current `master`:

`````` python
In [1]: import sympy

In [2]: f = sympy.Function('f')

In [3]: x = sympy.Symbol('x')

In [4]: sympy.dsolve(sympy.Eq(f(2 * x), sympy.sin(sympy.Derivative(f(x)))), f(x))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-b63e9ece1683> in <module>()
----> 1 sympy.dsolve(sympy.Eq(f(2 * x), sympy.sin(sympy.Derivative(f(x)))), f(x))

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/solvers/ode.pyc in dsolve(eq, func, hint, simplify, ics, xi, eta, x0, n, **kwargs)
    652             # The key 'hint' stores the hint needed to be solved for.
    653             hint = hints['hint']
--> 654             return _helper_simplify(eq, hint, hints, simplify)
    655 
    656 def _helper_simplify(eq, hint, match, simplify=True, **kwargs):

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/solvers/ode.pyc in _helper_simplify(eq, hint, match, simplify, **kwargs)
    674         # attempt to solve for func, and apply any other hint specific
    675         # simplifications
--> 676         sols = solvefunc(eq, func, order, match)
    677         free = eq.free_symbols
    678         cons = lambda s: s.free_symbols.difference(free)

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/solvers/ode.pyc in ode_lie_group(eq, func, order, match)
   5373                         if sep:
   5374                             # Trying to separate, r and s coordinates
-> 5375                             deq = integrate((1/sep[s]), s) + C1 - integrate(sep['coeff']*sep[r], r)
   5376                             # Substituting and reverting back to original coordinates
   5377                             deq = deq.subs([(r, rcoord), (s, scoord)])

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/utilities/decorator.pyc in threaded_func(expr, *args, **kwargs)
     33                                       func(expr.rhs, *args, **kwargs))
     34             else:
---> 35                 return func(expr, *args, **kwargs)
     36 
     37     return threaded_func

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/integrals/integrals.pyc in integrate(*args, **kwargs)
   1230     if isinstance(integral, Integral):
   1231         return integral.doit(deep=False, meijerg=meijerg, conds=conds,
-> 1232                              risch=risch, manual=manual)
   1233     else:
   1234         return integral

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/integrals/integrals.pyc in doit(self, **hints)
    485                     function, xab[0],
    486                     meijerg=meijerg1, risch=risch, manual=manual,
--> 487                     conds=conds)
    488                 if antideriv is None and meijerg1 is True:
    489                     ret = try_meijerg(function, xab)

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/integrals/integrals.pyc in _eval_integral(self, f, x, meijerg, risch, manual, conds)
    892                             result = result.func(*[
    893                                 arg.doit(manual=False) if arg.has(Integral) else arg
--> 894                                 for arg in result.args
    895                             ]).expand(multinomial=False,
    896                                       log=False,

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/core/basic.pyc in doit(self, **hints)
   1505         if hints.get('deep', True):
   1506             terms = [ term.doit(**hints) if isinstance(term, Basic) else term
-> 1507                                          for term in self.args ]
   1508             return self.func(*terms)
   1509         else:

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/integrals/integrals.pyc in doit(self, **hints)
    399         function = self.function
    400         if deep:
--> 401             function = function.doit(**hints)
    402         if function.is_zero:
    403             return S.Zero

/usr/lib/python2.7/site-packages/sympy-0.7.6-py2.7.egg/sympy/core/basic.pyc in doit(self, **hints)
   1505         if hints.get('deep', True):
   1506             terms = [ term.doit(**hints) if isinstance(term, Basic) else term
-> 1507                                          for term in self.args ]
   1508             return self.func(*terms)
   1509         else:

TypeError: doit() got an unexpected keyword argument 'manual'```
``````

### PR DESCRIPTION
**Pull request by mohitacecode**
#### References to other Issues or PRs

Fixes #9446

#### Brief description of what is fixed or changed
Added test for dsolve TypeError: doit() in `test_ode.py`.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/solvers/ode/tests/test_ode.py [hunk 1]
```diff
@@ -4073,3 +4073,9 @@ def test_issue_18408():
     sol = Eq(f(x), C1 + C3*exp(-x) + x*sinh(x)/2 + (C2 + x/2)*exp(x))
     assert sol == dsolve(eq, hint='nth_linear_constant_coeff_undetermined_coefficients')
     assert checkodesol(eq, sol) == (True, 0)
+
+
+def test_issue_9446():
+    f = Function('f')
+    assert dsolve(Eq(f(2 * x), sin(Derivative(f(x)))), f(x)) == \
+    [Eq(f(x), C1 + Integral(asin(f(2*x)), x)), Eq(f(x), C1 + pi*x - Integral(asin(f(2*x)), x))]
```
Comment (czgdp1807) [inline sympy/solvers/ode/tests/test_ode.py]: Does the order remains same on every run?

Comment (mohitacecode) [reply sympy/solvers/ode/tests/test_ode.py]: Yes It seems so.

**Comment by mohitacecode** [comment]
I have added the test please have a look If you have any suggestion @oscarbenjamin .

**Comment by oscarbenjamin** [comment]
I have commented on #9446 that the original problem was caused by an integral and which PR it was fixed in. There should be a test for that integral. Can you look through the PR that fixed the issue and see whether any tests are already added for this?

**Comment by mohitacecode** [comment]
Sure I will look and will add the test.

**Comment by mohitacecode** [comment]
I have added the required test from  #9446.

**Comment by oscarbenjamin** [comment]
This looks good. Merging.

**Pull request merged**